FlowCrypt/Controllers/Setup/SetupGenerateKeyViewController.swift

```diff
@@ -144,27 +147,42 @@ private actor Service {
                            for: user.email)
 
         if storageMethod == .memory {
-            let passPhrase = PassPhrase(value: passPhrase, fingerprintsOfAssociatedKey: encryptedPrv.key.fingerprints)
+            let passPhrase = PassPhrase(
+                value: passPhrase,
+                fingerprintsOfAssociatedKey: encryptedPrv.key.fingerprints
+            )
             passPhraseService.savePassPhrase(with: passPhrase, storageMethod: .memory)
         }
 
-        let updateKey = attester.updateKey(
+        await submitKeyToAttesterAndShowAlertOnFailure(
             email: userId.email,
-            pubkey: encryptedPrv.key.public,
-            token: storage.token
+            publicKey: encryptedPrv.key.public,
+            viewController: viewController
         )
 
-        try awaitPromise(await viewController.alertAndSkipOnRejection(
-            updateKey,
-            fail: "Failed to submit Public Key")
-        )
-        let testWelcome = attester.testWelcome(email: userId.email, pubkey: encryptedPrv.key.public)
-        try awaitPromise(await viewController.alertAndSkipOnRejection(
-            testWelcome,
-            fail: "Failed to send you welcome email")
+        _ = try await attester.testWelcome(
+            email: userId.email,
+            pubkey: encryptedPrv.key.public
         )
```

Comment:
I suppose this way if this throws, the whole async function will fail? As I mentioned earlier, in this one case I'd prefer a `try?`, including a // comment I mentioned earlier above, explaining why we chose `try?` this time.